cmd/staticlint/main.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"go/ast"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/multichecker"
+	"golang.org/x/tools/go/analysis/passes/asmdecl"
+	"golang.org/x/tools/go/analysis/passes/assign"
+	"golang.org/x/tools/go/analysis/passes/atomic"
+	"golang.org/x/tools/go/analysis/passes/atomicalign"
+	"golang.org/x/tools/go/analysis/passes/bools"
+	"golang.org/x/tools/go/analysis/passes/buildtag"
+	"golang.org/x/tools/go/analysis/passes/composite"
+	"golang.org/x/tools/go/analysis/passes/copylock"
+	"golang.org/x/tools/go/analysis/passes/deepequalerrors"
+	"golang.org/x/tools/go/analysis/passes/errorsas"
+	"golang.org/x/tools/go/analysis/passes/httpresponse"
+	"golang.org/x/tools/go/analysis/passes/ifaceassert"
+	"golang.org/x/tools/go/analysis/passes/loopclosure"
+	"golang.org/x/tools/go/analysis/passes/lostcancel"
+	"golang.org/x/tools/go/analysis/passes/nilfunc"
+	"golang.org/x/tools/go/analysis/passes/nilness"
+	"golang.org/x/tools/go/analysis/passes/printf"
+	"golang.org/x/tools/go/analysis/passes/shift"
+	"golang.org/x/tools/go/analysis/passes/stdmethods"
+	"golang.org/x/tools/go/analysis/passes/stringintconv"
+	"golang.org/x/tools/go/analysis/passes/structtag"
+	"golang.org/x/tools/go/analysis/passes/tests"
+	"golang.org/x/tools/go/analysis/passes/unmarshal"
+	"golang.org/x/tools/go/analysis/passes/unreachable"
+	"golang.org/x/tools/go/analysis/passes/unsafeptr"
+	"golang.org/x/tools/go/analysis/passes/unusedresult"
+	"golang.org/x/tools/go/analysis/passes/unusedwrite"
+	"honnef.co/go/tools/simple"
+	"honnef.co/go/tools/staticcheck"
+	"strings"
+)
+
+// osExitAnalyzer - Custom analyzer to prohibit os.Exit call in main func
+var osExitAnalyzer = &analysis.Analyzer{
+	Name: "reportOsExit",
+	Doc:  "check for os.Exit calls in main function of main package",
+	Run:  run,
+}
+
+// run parses the source file and defines a custom visitor to check for os.Exit in the main function.
+//
+// Parameter: pass *analysis.Pass
+// Return: (interface{}, error)
+func run(pass *analysis.Pass) (interface{}, error) {
+	// Get the main package
+	pkg := pass.Pkg
+	if pkg.Name() != "main" {
+		return nil, nil // Skip if it's not the main package
+	}
+
+	for _, file := range pass.Files {
+		// Check if it's the main function
+		for _, decl := range file.Decls {
+			fn, isFunc := decl.(*ast.FuncDecl)
+			if isFunc && fn.Name.Name == "main" {
+				// Inspect the function body for os.Exit calls
+				ast.Inspect(fn.Body, func(n ast.Node) bool {
+					callExpr, isCallExpr := n.(*ast.CallExpr)
```

Comment:
Anyway, it would be better if you created a new function.
It is not readable now.